### PR TITLE
Redo implementation of `mypy package` and `mypy directory`

### PR DESCRIPTION
**Pull request by gvanrossum**
There are unit tests too. Discussion in #1449.

### CODE REVIEW

**Comment by gvanrossum** [comment]
It's ready for review. If I don't hear from anyone I'll merge later today.

File: mypy/test/testcmdline.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from mypy.test.helpers import assert_string_arrays_equal
+
+# Path to Python 3 interpreter
+python3_path = sys.executable
+
+# Files containing test case descriptions.
+cmdline_files = ['cmdline.test']
+
+
+class PythonEvaluationSuite(Suite):
```
Comment (JukkaL) [inline mypy/test/testcmdline.py]: Maybe we should modify `runtests.py` to not consider these unit tests, as running mypy as a subprocess is slow. See `add_myunit` and `add_pythoneval` in `runtests.py`. I suggest special casing these to be considered evaluation test cases (though the name is not accurate) or adding a new category.

File: mypy/main.py [hunk 2]
```diff
@@ -275,33 +283,98 @@ def parse_version(v):
...
+                f = os.path.join(root, f)
+                if f.endswith(ext):
+                    mod = make_mod(top_dir, top_mod, f, ext)
+                    if mod not in mods:
+                        sources.append(BuildSource(f, mod, None))
+                        mods.add(mod)
+    return sources
+
+
+def make_mod(dir: str, mod: str, file: str, ext: str) -> str:
```
Comment (JukkaL) [inline mypy/main.py]: `mod` is unused

**Comment by JukkaL** [comment]
Did a *very* quick pass and seems reasonable to me, except for a few small things (see the notes).

**Pull request merged**